### PR TITLE
Fix bug1086 resizing side panel causes crash

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This PR fixes issue #1086, by preventing the stack overflow.

This section is copied from the third commit.

```Several infinitely recursive loops passing through
SidePanel::PlanetPanelContainer::DoPanelsLayout() were possible.

In one PlanetPanelContainer::DoPanelsLayout() would call
PlanetPanel::DoLayout(), which would signal ResizedSignal(), which would
call DoPanelsLayout() in an infinite loop.

A second loop passed through AccordionPanel::SetCollapsed(), which even
when the collapsed state was unchanged still signaled with the
ExpandCollapseSignal().
```

The third commit is possibly the only commit necessary to stop SidePanel having a stack overflow.

The second commit prevents AccordionPanel from doing unnecessary work and potentially causing other unnecessary work.

The first commit is required since currently SidePanel expects the unnecessary signal from AccordionPanel before it correctly sizes the focus drop down icons.

### CODE REVIEW

File: UI/SidePanel.cpp [hunk 1]
```diff
@@ -606,6 +606,8 @@ class SidePanel::PlanetPanelContainer : public GG::Wnd {
                                 m_valid_selection_predicate;
 
     GG::Scroll*                 m_vscroll; ///< the vertical scroll (for viewing all the planet panes)
+
+    bool                        m_ignore_recurisve_resize;
```
Comment (Vezzra) [inline UI/SidePanel.cpp]: Typo: recurisve => recursive

**Review by Vezzra — CHANGES_REQUESTED** [review]

**Comment by geoffthemedio** [comment]
otherwise ok

**Comment by LGM-Doyle** [comment]
I've fixed the spelling error.

**Review by Vezzra — APPROVED** [review]

**Comment by LGM-Doyle** [comment]
Thanks for the review.

**Pull request merged**